integrationtests/src/test/java/org/axonframework/integrationtests/testsuite/student/AbstractStudentTestSuite.java

```diff
@@ -58,25 +59,30 @@ public abstract class AbstractStudentTestSuite {
 
     protected static final GenericCommandResultMessage<String> SUCCESSFUL_COMMAND_RESULT =
             new GenericCommandResultMessage<>(new MessageType("empty"), "successful");
-
-    private StatefulCommandHandlingModule.CommandHandlerPhase statefulCommandHandlingModule;
-    private EventSourcedEntityBuilder<String, Course> courseEntity;
-    private EventSourcedEntityBuilder<String, Student> studentEntity;
-
     protected CommandGateway commandGateway;
-    protected Repository<String, Student> studentRepository;
-    protected Repository<String, Course> courseRepository;
+    protected StateManager stateManager;
+    private StatefulCommandHandlingModule.CommandHandlerPhase statefulCommandHandlingModule;
+    private EventSourcedEntityModule<String, Course> courseEntity;
+    private EventSourcedEntityModule<String, Student> studentEntity;
 
     @BeforeEach
     void setUp() {
-        studentEntity = EventSourcedEntityBuilder.entity(String.class, Student.class)
-                                                 .entityFactory(c -> EventSourcedEntityFactory.fromIdentifier(Student::new))
-                                                 .criteriaResolver(this::studentCriteriaResolver)
-                                                 .entityEvolver(this::studentEvolver);
-        courseEntity = EventSourcedEntityBuilder.entity(String.class, Course.class)
-                                                .entityFactory(c -> EventSourcedEntityFactory.fromIdentifier(Course::new))
-                                                .criteriaResolver(this::courseCriteriaResolver)
-                                                .entityEvolver(this::courseEvolver);
+        studentEntity = EventSourcedEntityModule
+                .declarative(String.class, Student.class)
+                .entityModel(c -> EntityModel.forEntityType(Student.class)
```

Comment:
Just a check from my end: we're now setting an `EntityModel` on the declarative `EventSourcedEntityModule` so that we do not have to duplicate **all** the logic that resides on the `EntityModel` for the `EventSourcedEntityModule`, right?

Looking for justification of a builder in a builder, recalling that you discussed this with me. Just doing an additional check.

Reply:
Yes, you are correct. There are many types of entity models, and supporting them all across various builders would be less than optimal.

To ease it for the user, the function now takes (Configuration, EntityMetamodelBuilder<E>), so they don't have to know of the different builder implementations for the most common usecase